lib/addons/amp/index.ts

```diff
@@ -0,0 +1,131 @@
+    workspaceName: 'blueprints-amp-workspace',
+    workspaceTagKey: 'Project',
+    workspaceTagValue: 'blueprints-amp-workspace',
+    deploymentMode: DeploymentMode.DEPLOYMENT
+};
+
+/**
+ * Implementation of AMP add-on for EKS Blueprints. Installs ADOT Collector.
+ */
+export class AmpAddOn implements ClusterAddOn {
+
+    readonly ampAddOnProps: AmpAddOnProps;
+
+    constructor(props?: AmpAddOnProps) {
+        this.ampAddOnProps = { ...defaultProps, ...props };
+    }
+
+    @dependable(AdotCollectorAddOn.name)
+    deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+        const cluster = clusterInfo.cluster;
+        let finalRemoteWriteURLEndpoint: string;
+        let finalDeploymentMode: string;
+        let doc: string;
+        
+        let cfnWorkspaceProps : CfnWorkspaceProps  = {
+            alias: defaultProps.workspaceName,  
+            tags: [{
+            key: defaultProps.workspaceTagKey,
+            value: defaultProps.workspaceTagValue
+            }],
+        }
+
+        if (this.ampAddOnProps.workspaceName){
+            cfnWorkspaceProps = {
+                ...cfnWorkspaceProps,
+                alias:this.ampAddOnProps.workspaceName
+            };        
+        }
+
+        if (this.ampAddOnProps.workspaceTagKey && this.ampAddOnProps.workspaceTagValue){
+            cfnWorkspaceProps = {
+                ...cfnWorkspaceProps,
+                tags: [{
+                    key: this.ampAddOnProps.workspaceTagKey,
+                    value: this.ampAddOnProps.workspaceTagValue
+                }]
+            }
+        }
+
+        if (this.ampAddOnProps.prometheusRemoteWriteURL) {
+            finalRemoteWriteURLEndpoint = this.ampAddOnProps.prometheusRemoteWriteURL;
+        }
+        else {
+            const cfnWorkspace = new aps.CfnWorkspace(cluster.stack, 'MyAMPWorkspace', cfnWorkspaceProps);/* all optional props */ 
+            const URLEndpoint = cfnWorkspace.attrPrometheusEndpoint;
+            finalRemoteWriteURLEndpoint = URLEndpoint + 'api/v1/remote_write';
+        }
+
+        finalDeploymentMode = defaultProps.deploymentMode
+        if (this.ampAddOnProps.deploymentMode) {
+            finalDeploymentMode = this.ampAddOnProps.deploymentMode
+        }
+
+        // Applying manifest for configuring ADOT Collector for Amp.
+        if (finalDeploymentMode == DeploymentMode.DAEMONSET) {
+            doc = readYamlDocument(__dirname + '/collector-config-amp-daemonset.ytpl');
+        }
+        else {
+            doc = readYamlDocument(__dirname + '/collector-config-amp.ytpl');
+        }
+        const docArray = doc.replace(/{{your_remote_write_endpoint}}/g, finalRemoteWriteURLEndpoint)
+        .replace(/{{your_aws_region}}/g, cluster.stack.region)
+        .replace(/{{your_deployment_method}}/g, finalDeploymentMode);
+        const manifest = docArray.split("---").map(e => loadYaml(e));
+        const statement = new KubernetesManifest(cluster.stack, "adot-collector-amp", {
```

Comment:
sounds like there is a missing dependency between cfn workspace and this manifest. It would make sense to make sure that workspace is deployed, otherwise I assume the yaml will go first always and create at warnings, may also fail since URL wont' be functioning yet.

Reply:
Im not able to add a dependency for `cfnWorkspace` type to node dependency. I wil connect with you on this.